plash_cli/core.py

```diff
@@ -106,23 +97,32 @@ def _deps(script: bytes | str) -> dict | None:
         return '\n'.join(tomllib.loads(content)['dependencies'])
     else: return None
 
-# %% ../nbs/00_core.ipynb 15
-def create_tar_archive(path)->io.BytesIO: # Buffer of tar directory
-    "Creates a tar archive of a directory, excluding files based on is_included"
+# %% ../nbs/00_core.ipynb 14
+def prepare_archive(path: Path) -> tuple[io.BytesIO, int]:
+    """Validates app structure and creates a tar archive."""
+    if not (path / 'main.py').exists():
+        print('ERROR: Supplied directory is invalid. A Plash app requires a main.py file.')
+        print(f'Invalid path: {path}')
+        sys.exit(1)
+        
+    deps = _deps((path / 'main.py').read_text())
+    if not (bool(deps) ^ (path/"requirements.txt").exists()):
```

Comment:
What about the the case where neither exist? I feel it should be fine if their app contains no third-party dependencies

Reply:
python-fasthtml should always be in there right? it's not included by default anymore.